### ISSUE TITLE
Add Support for Synchronous Calls

### ISSUE THREAD

**Issue by daryllabar** [SimplerSoftware/XrmTSToolkit#3]
I'm attempting calling a Soap Call from a call back from an Alert.Js Callback method.  Unfortunately, I need to be able to perform a synchronous call, rather than an async call.  Can we add an optional parameter to the Calls to be sync (they can be async by default, unless specified)?

**Comment by daryllabar** [comment]
I've implemented the simplest change for this possible: https://github.com/SimplerSoftware/XrmTSToolkit/pull/4

**Comment by StevenRasmussen** [comment]
I'll take a look and see what makes sense.  We can definitely add in support for synchronous calls but with TypeScript we can only really have one type of return value from any single method, which is currently a jQuery promise.  So we may just need to create separate 'Synchronous' versions of each of the major messages which would return the appropriate responses from CRM.

**Comment by daryllabar** [comment]
You could create an extra "async" parameter for all calls, that defaults to true, so no current usages of it changes:

``` js
        static Create(entity: Soap.Entity, async: true): JQueryPromise<Soap.CreateSoapResponse> {
            var requestBody = "<entity>" + entity.Serialize() + "</entity>";
            return $.Deferred<Soap.CreateSoapResponse>(function (dfd) {
                var request = Soap.DoRequest<Soap.CreateSoapResponse>(requestBody, "Create", async);
                request.done(function (data, result, xhr) {
                    dfd.resolve(data);
                });
                request.fail(function (result) {
                    dfd.reject(result);
                });
            }).promise();
        }

...

private static DoRequest<T>(soapBody: string, requestType: string, async: boolean): JQueryPromise<T> {
    var xml = "";
    if (requestType == "Execute") {
        xml = "<s:Envelope xmlns:s = \"http://schemas.xmlsoap.org/soap/envelope/\">" +
            "<s:Body>" + soapBody + "</s:Body></s:Envelope>";
    }
    else {
        soapBody = "<" + requestType + ">" + soapBody + "</" + requestType + ">";
        //Add in all the different namespaces
        xml = "<s:Envelope" + Soap.GetNameSpacesXML() + "><s:Body>" + soapBody + "</s:Body></s:Envelope>";
    }

    return $.Deferred<T>(function (dfd) {
        var Request = $.ajax(XrmTSToolkit.Common.GetSoapServiceURL(), {
            async: async,
            data: xml,
            type: "POST",
            beforeSend: function (xhr: JQueryXHR) {
                xhr.setRequestHeader("Accept", "application/xml, text/xml, */*");
                xhr.setRequestHeader("Content-Type", "text/xml; charset=utf-8");
                xhr.setRequestHeader("SOAPAction", "http://schemas.microsoft.com/xrm/2011/Contracts/Services/IOrganizationService/" + requestType);
            }
        });
        Request.done(function (data: any, textStatus: string, xhr: JQueryXHR) {
            var sr: Soap.SoapResponse = null;
            var rt = xhr.responseText;
            switch (requestType) {
                case "Create":
                    sr = new Soap.CreateSoapResponse(rt);
                    break;
                case "Update":
                    sr = new Soap.UpdateSoapResponse(rt);
                    break;
                case "Delete":
                    sr = new Soap.DeleteSoapResponse(rt);
                    break;
                case "Retrieve":
                    sr = new Soap.RetrieveSoapResponse(rt);
                    break;
                case "RetrieveMultiple":
                    sr = new Soap.RetrieveMultipleSoapResponse(rt);
                    break;
                case "Execute":
                    sr = new Soap.ExecuteResponse(rt);
                    break;
                default:
                    sr = new Soap.SoapResponse(rt);
                    break;
            }
            sr.ParseResult();
            dfd.resolve(<T><any>sr, textStatus, xhr);
        });
        Request.fail(function (xhr: JQueryXHR, textStatus: string, exception) {
            var errorText = typeof exception === "undefined" ? textStatus : exception;
            var fault = new Soap.FaultResponse(xhr.responseText);
            fault.ParseResult();
            dfd.reject(fault);
        });
    }).promise();
}
```

**Comment by StevenRasmussen** [comment]
This was addressed in v0.8.5
